src/sentry/profiles/task.py

```diff
@@ -144,10 +155,11 @@ def process_profile_task(
         set_measurement("profile.stacks.processed", len(profile["profile"]["stacks"]))
         set_measurement("profile.frames.processed", len(profile["profile"]["frames"]))
 
-    if options.get(
-        "profiling.generic_metrics.functions_ingestion.enabled"
-    ) and project.organization_id in options.get(
-        "profiling.generic_metrics.functions_ingestion.allowed_org_ids"
+    if (
+        profile.get("version") != "2"
```

Comment:
For clarity, can we put this in a helper function? Perhaps something like

```python
def get_profile_version(profile):
  version = profile.get("version")
  if version == "2":
    return 2
  # older SDKs may not set this value so assume version 1
  return 1
```

I assume it's `!= "2"` because the empty string absence of implies version 1 in some older sdks.

Same for all the calls later.

Reply:
No version actually implies the `legacy` format.